### PR TITLE
Move shallow state logic to the function preparing state for workers

### PR DESCRIPTION
**Pull request by maximedenes**
This patch turns the three-valued `marshallable` type into a boolean. The previous `Shallow` value is now handled directly by the code preparing the state for workers, rather than by clients of the `Summary` API.

The type `marshallable` was used for several different purposes (see the mapping from `Yes to `No ...), so I hope this makes the code clearer.

We may be able to remove the `marshallable` flag altogether, at some point.

### CODE REVIEW

**Comment by gares** [comment]
> We may be able to remove the `marshallable` flag altogether, at some point.

Agreed. The introduction of the flag was before the ephemeron mechanism was added.
This latter mechanism supersedes the `~marshallable` flag (after this PR).

**Review by gares — APPROVED** [review]

**Comment by gares** [comment]
I'm afraid you'll have to prepare an overlay for any plugin that registers a piece of state.
And do it again to remove marshall... maybe you should do both ;-)

**Comment by maximedenes** [comment]
> I'm afraid you'll have to prepare an overlay for any plugin that registers a piece of state.
> And do it again to remove marshall... maybe you should do both ;-)

Hmm, I did so already. It broke only paramcoq, in very exotic piece of code (the comment just above reads "I have no idea what I'm doing").

This is ready for merge, modulo maybe AppVeyor.

**Review by ejgallego — APPROVED** [review]

**Comment by maximedenes** [comment]
@gares this is ready for merge

File: stm/stm.ml [hunk 1]
```diff
@@ -2133,14 +2136,14 @@ end (* }}} *)
...
-    doc:doc -> ?redefine_qed:bool -> cache:Summary.marshallable ->
+    doc:doc -> ?redefine_qed:bool -> cache:bool ->
       Stateid.t -> unit
 
 end = struct (* {{{ *)
 
 let async_policy () =
-  if Attributes.is_universe_polymorphism () then false
-  else if VCS.is_interactive () = `Yes then
+  if Attributes.is_universe_polymorphism () then false (* FIXME this makes no sense, it is the default value of the attribute *)
```
Comment (SkySkimmer) [inline stm/stm.ml]: @maximedenes what did you mean with this comment?

**Pull request merged**